contracts/Staking.sol

```diff
@@ -78,11 +81,11 @@ contract Staking is Ownable {
             amount = msg.value;
             tokenAddress = address(0);
         } else {
-            require(IERC20(tokenAddress).transferFrom(msg.sender, address(this), amount), "Staking: could not transfer tokens");
```

Comment:
Why do you remove this require?

Reply:
The security recommendation was:  **"in function unstake(): add require() around IERC20 transfer() call as performed in stakeFor() for example. Also, consider using SafeERC20 [refs: https://soliditydeveloper.com/safe-erc20, https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#SafeERC20"**

I replaced the validation by using a contract provided by OZ named `SafeERC20` which is used as a library and already has this validation. I guess its just a different way of doing it.

Reply:
Then I did the same thing for `stake` just to be consistent.

Reply:
Clear, but then should we also use safeTransfer in the Storage Manager contract?

Reply:
I think the StorageManager contract cannot take any more code. Its quite big already, thats why i kept the previous approach in this one. No margin for sugar!